text/0240-app-telemetry-schema-vision-roadmap.md

```diff
@@ -0,0 +1,430 @@
+  server via a URL. This artifact must be small and avoid the repetition of
+  definitions.
+* **Easy to parse**: A widespread and well-defined format should be preferred.
+  JSON is an example of such a format.
+* **Easy to interpret**: The internal structure of this artifact must be
+  straightforward to avoid any misinterpretation.
+* **Platform- and Language-agnostic**: This artifact must be independent of any
+  platform architectures and programming languages.
+
+The following diagram describes two main use cases for the Resolved Telemetry
+Schema. The key points to remember are: 1) both use cases result in a Resolved
+Telemetry Schema, and 2) Resolved Telemetry Schemas serve as the mechanism for
+distributing Telemetry Schemas throughout the entire ecosystem.
+
+![Use cases](./img/0240-otel-weaver-use-cases.svg)
+
+The main components of a Resolved Telemetry Schema are illustrated in the
+diagram below. The 'OTel Weaver' tool is used to create these schemas. It can
+also extend an existing schema or import a Semantic Convention Registry.
+Resolved Telemetry Schema serves as a key mechanism for interoperability,
+feeding various external tools, including SDK generators, documentation
+generators, policy enforcers, and more.
+
+![Resolved Telemetry Schema](./img/0240-otel-weaver-resolved-schema.svg)
+
+The internal catalog is used to define all the attributes and metrics in this
+artifact. This design allows for the reuse of the same attributes or metrics
+multiple times in different signals and different instrumentation libraries. It
+is expected to be a very common pattern to reuse the same subset of attributes
+or metrics across several signals and libraries.
+
+A dedicated OTEP will precisely define the structure and format of the Resolved
+Telemetry Schema.
+
+### Development Strategies
+
+Two development strategies coexist and must be supported. The first strategy, a
+monorepo type (single source of truth), has complete control over the
+applications, their dependencies, and the associated telemetry schemas. The
+second strategy is more heterogeneous, composed of multiple repositories, where
+the build process of external dependencies is out of the control of the entity
+owning the applications or services.
+
+In the first model, each build process can independently apply telemetry schema
+policies in parallel, knowing that the policies are shared and the entire
+environment is controlled by the same entity.
+
+In the second model, the application or service environment does not control the
+telemetry schema policies of external dependencies. There is a need for a method
+to retrieve telemetry schemas from these dependencies. The mechanism for
+distributing these schemas is still being discussed (refer to the Open Questions
+section). Ultimately, this will enable the CI/CD pipeline of the application or
+service to apply its own local policies to these schemas from the dependencies.
```

Comment:
It's not clear to me how this should work. 

I guess this might work if those "local policies" only have an effect on the SDK, but I assume they can't change any API that is used by the dependencies?

Anyway, this is covered by an open question, and doesn't need to be answered in this OTEP.

Reply:
Indeed, this will be covered in a future OTEP.